tests/templates/test_subroutines/test_commuting_evolution.py

```diff
@@ -51,6 +51,25 @@ def evolution_circuit(time):
     assert all(np.isclose(dev1.state, dev2.state))
 
 
+def test_decomposition_expand():

```

Comment:
It can be worth adding a test case similar to the example in the docstring too. That would double-check that the result of the computation itself would be correct.